src/vs/workbench/contrib/tasks/electron-browser/task.contribution.ts

```diff
@@ -1425,8 +1429,8 @@ class TaskService extends Disposable implements ITaskService {
 					}
 				}
 			}
-			return this.getWorkspaceTasks().then((customTasks) => {
-				customTasks.forEach((folderTasks, key) => {
+			return this.getWorkspaceTasks().then(async (customTasks) => {
+				await Promise.all(Array.from(customTasks).map(async ([key, folderTasks]) => {
```

Comment:
This Promise.all(Array.from(...).map(async ......)) is unnecessarily hard to read.

Reply:
Would you prefer some multi-line formatting or for these to be extracted into some local consts instead of one big expression?

It's a bummer having to add `Array.from` all over the place when I want to map an iterable.

Reply:
Went with local consts.